### ISSUE TITLE
Magento 2.2.3 TypeErrors Cannot read property 'quoteData' / 'storecode' / 'sectionLoadUrl' of undefined 

### ISSUE THREAD

**Issue by brvm** [magento/magento2#14412]
### Preconditions

1. Magento 2.2.3
2. PHP 7.0
3. MySQL 5.5.38

### Steps to reproduce

1. Create a module in app/code/vendor
2. Add checkout_cart_index.xml in templates folder under /view/frontend/layout
3. Add following code:
```
<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" layout="1column" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
    <head>
        <link src="Vendor_Module/js/jquery.cartAjaxQty.js"/>
    </head>
</page>
```
4. Add .js-file under /view/frontend/web/js called jquery.cartAjaxQty.js

```
require(['jquery',
    'domReady!',
    'Magento_Checkout/js/model/totals',
    'Magento_Checkout/js/view/summary/abstract-total',
    'Magento_Customer/js/customer-data',
    'Magento_Checkout/js/model/quote',
    'Magento_Catalog/js/price-utils',
    'Magento_Checkout/js/action/get-totals'], function () {
    jQuery(document).ready(function($){

        $(".cart-container .form-cart .action.update").remove();
        jQuery('#shopping-cart-table')
            .on(
                'change',
                'input[name$="[qty]"]',
                function(){
                    console.log("quantity changed");
                    var form = jQuery(jQuery(this).closest('form'));

                    $.ajax({
                        url: form.attr('action'),
                        data: form.serialize(),
                        type: 'post',
                        showLoader: true,
                        context: this.changedInput,
                        success: function (res) {
                            // var parsedResponse = $.parseHTML(res);
                            // var result = $(parsedResponse).find("#form-validate");
                            // $("#form-validate").replaceWith(result);
                            //
                            console.log('Ajax succes function completed');
                        },
                        error: function (xhr, status, error) {
                            var err = eval("(" + xhr.responseText + ")");
                            console.log(err.Message);
                        }
                    });

                }
            );
    });
});
```
5. Add .js file to requirejs-config.js located in root frontend folder in module under /view/frontend

```
var config = {
    map: {
        '*': {
            ajaxQty: 'Vendor_Module/js/jquery.cartAjaxQty'
        }
    }
};
```
6. Run

`php bin/magento setup:upgrade`

7. Add item to cart and change the quantity in the input box on the cart page.

### Expected result

1. The .js code should load and requirements should be loaded properly
2. When quantity changes it should output this in the console

(and later on I want to add code to reload the summary, that's why there are all those requirements in the require[])

### Actual result

1. Three different errors (sometimes displaying all in console, mostly one or two of the three) caused by most of the requirements:

![screen shot 2018-03-28 at 11 43 54](https://user-images.githubusercontent.com/19427937/38021569-8253ea00-327d-11e8-9caf-faaeb71fed45.png)
![screen shot 2018-03-28 at 11 44 19](https://user-images.githubusercontent.com/19427937/38021570-826c2e1c-327d-11e8-9f9b-04edd91bb9ba.png)
![screen shot 2018-03-28 at 11 44 28](https://user-images.githubusercontent.com/19427937/38021571-828785ea-327d-11e8-8356-25eb2d44787b.png)

**Comment by swayoleg** [comment]
my workaroud was 
1) create quote.js at design package
2) add     'domReady!'


copy vendor/magento/module-checkout/view/frontend/web/js/model/quote.js 
to 
app/design/frontend/%YOUR_DESIGNVENDOR%/%themename%/Magento_Checkout/web/js/model/quote.js

Then change define to 

 `define([
    'ko',
    'underscore',
    'domReady!'
],` 
(add domReady!)

But its just a workaround. We need a core fix

**Comment by brvm** [comment]
We do. I have made a Stack topic about it and eventually found a (temporary) fix:

https://magento.stackexchange.com/questions/220206/dependecy-error-magento-2-sectionloadurl-error

**Comment by renttek** [comment]
I also encountered this bug today, but only in Chrome and only for `quoteData`. THe workaround seems to fix this

**Comment by LiamKarlMitchell** [comment]
A possibly similar issue with Magento ver. 2.2.0
Installed extension if it matters: Free Gift for Magento 2

```Cannot read property `quoteData` of undefined at quote.js:32```
![image](https://user-images.githubusercontent.com/1968109/40149677-17867896-59c9-11e8-8050-8bbb4937e53c.png)

**Comment by Notall-wo** [comment]
I also encountered this bug,shield the'Magento_Checkout/js/action/get-totals'  will not report an error, but there aren't data. don't know why.

**Comment by Notall-wo** [comment]
I also encountered this bug, shield the'Magento_Checkout/js/action/get-totals' will not report an error,  I don't know why it's affected, so no data was loaded

**Comment by ghost** [comment]
Hi @bramulous , thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by magento-engcom-team** [comment]
Hi @bramulous. Thank you for your report. 
The issue has been fixed in magento/magento2#18503 by @ihor-sviziev in 2.3-develop branch
Related commit(s):
 - [96687ac49b311ababf2d3eb7054b07cf9627883d](https://github.com/magento/magento2/commit/96687ac49b311ababf2d3eb7054b07cf9627883d)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @bramulous. Thank you for your report. 
The issue has been fixed in magento/magento2#21432 by @amol2jcommerce in 2.2-develop branch
Related commit(s):
 - [d40ea45a5827fd4da763d20be6cde0d3b279809f](https://github.com/magento/magento2/commit/d40ea45a5827fd4da763d20be6cde0d3b279809f)

The fix will be available with the upcoming 2.2.9 release.
